smoke-test/tests/cypress/cypress/e2e/search/query_and_filter_search.js

```diff
@@ -1,57 +1,97 @@
-describe("auto-complete dropdown, filter plus query search test", () => {
-
-  const platformQuerySearch = (query,test_id,active_filter) => {
-    cy.visit("/");
-    cy.get("input[data-testid=search-input]").type(query);
-    cy.get(`[data-testid="quick-filter-urn:li:dataPlatform:${test_id}"]`).click();
-    cy.focused().type("{enter}").wait(3000);
-    cy.url().should(
-      "include",
-      `?filter_platform___false___EQUAL___0=urn%3Ali%3AdataPlatform%3A${test_id}`
-    );
-    cy.get('[data-testid="search-input"]').should("have.value", query);
-    cy.get(`[data-testid="active-filter-${active_filter}"]`).should("be.visible");
-    cy.contains("of 0 results").should("not.exist");
-    cy.contains(/of [0-9]+ results/);
-  }
-
-  const entityQuerySearch = (query,test_id,active_filter) => {
-    cy.visit("/");
-    cy.get("input[data-testid=search-input]").type(query);
-    cy.get(`[data-testid="quick-filter-${test_id}"]`).click();
-    cy.focused().type("{enter}").wait(3000);
-    cy.url().should(
-      "include",
-      `?filter__entityType___false___EQUAL___0=${test_id}`
-    );
-    cy.get('[data-testid="search-input"]').should("have.value", query);
-    cy.get(`[data-testid="active-filter-${active_filter}"]`).should("be.visible");
-    cy.contains("of 0 results").should("not.exist");
-    cy.contains(/of [0-9]+ results/);
-  }
+const datasetNames = {
+  dashboardsType: "Baz Dashboard",
+  pipelinesType: "Users",
+  MlmoduleType: "scienceModel",
+  glossaryTermsType: "AccountBalance",
+  tags: "Baz Chart 1",
+  hivePlatform: "datahub_db",
+  airflowPlatform: "User Creations",
+  awsPlatform: "project/root/events/logging_events_bckp",
+  hdfsPlatform: "SampleHdfsDataset"
+};
 
+const searchToExecute = (value) => {
+  cy.get("input[data-testid=search-input]").eq(0).type(`${value}{enter}`);
+  cy.waitTextPresent("Type");
+};
+
+const selectFilteredEntity = (textToClick, entity, url) => {
+  cy.get(`[data-testid=filter-dropdown-${textToClick}]`).click({ force: true });
+  cy.get(`[data-testid="filter-option-${entity}"]`).click({ force: true });
+  cy.get("[data-testid=update-filters]").click({ force: true });
+  cy.url().should("include", `${url}`);
+  cy.get("[data-testid=update-filters]").should("not.be.visible");
+  cy.get('.ant-pagination-next').scrollIntoView().should('be.visible');
+};
+
+const verifyFilteredEntity = (text) => {
+  cy.get('.ant-typography').contains(text).should('be.visible');
+};
+
+ describe("auto-complete dropdown, filter plus query search test", () => {
   it("verify the 'filter by' section + query (result in search page with query applied + filter applied)", () => {
```

Comment:
I believe this should not be a single test but multiple tests

Reply:
okay I will split it into multiple tests